### PR TITLE
Withdrawals

### PR DESCRIPTION
**Pull request by debjit-bw**
adds test for withdrawals:
* gets withdrawals root for a specific withdrawals list (from both reth and nethermind)
* asserts equality

For reth, we first deploy the deposit contract address

### CODE REVIEW

File: scripts/test_withdrawals_eip4895.sh [hunk 1]
```diff
@@ -0,0 +1,483 @@
+#!/bin/bash
```
Comment (dapplion) [inline scripts/test_withdrawals_eip4895.sh:1]: Adding separate bash scripts for every test scales pretty bad. The scripts should result in producing a sequence of blocks from Nethermind with the engine API and then apply those blocks on Reth.

So you can have multiple generators, each generates a directory of block JSONs `blocks_test_withdrawals_eip4895`, then we have a script that you pass a `blocks_` directory as argument and runs the test.

File: src/payload_builder.rs [hunk 2]
```diff
@@ -466,15 +437,37 @@ where
     .map_err(|err| PayloadBuilderError::Internal(err.into()))?;
     // GNOSIS SPECIFIC >
 
+    // calculate the requests and the requests root
+    let requests = if chain_spec.is_prague_active_at_timestamp(attributes.timestamp) {
+        let deposit_requests = parse_deposits_from_receipts(&chain_spec, receipts.iter().flatten())
+            .map_err(|err| PayloadBuilderError::Internal(RethError::Execution(err.into())))?;
+
+        println!(
```
Comment (dapplion) [inline src/payload_builder.rs:445]: Can't merge this log

File: scripts/test_withdrawals_eip4895.sh (hunk 1 above)
Comment (debjit-bw) [reply scripts/test_withdrawals_eip4895.sh:1]: that's a better idea, i will close both PRs and do this instead

File: src/payload_builder.rs (hunk 2 above)
Comment (debjit-bw) [reply src/payload_builder.rs:445]: my bad, removing

**Comment by dapplion** [comment]
Closing in favor of:

> Adding separate bash scripts for every test scales pretty bad. The scripts should result in producing a sequence of blocks from Nethermind with the engine API and then apply those blocks on Reth.

> So you can have multiple generators, each generates a directory of block JSONs blocks_test_withdrawals_eip4895, then we have a script that you pass a blocks_ directory as argument and runs the test.

**Pull request closed without merging**